projects/reservoir/fv3/era_process.py

```diff
@@ -37,18 +62,34 @@ def main(args):
     setup_fregrid()
     for var in variables:
         check_time_steps_complete(path, var)
-        weekdays = ["mon", "tue", "wed", "thu", "fri", "sat", "sun"]
-        for i, day in enumerate(weekdays):
-            merge_into_file(path, var, i, day)
-            logger.info(f"{day} is merged")
-            regrid_to_cubed_sphere(path, var, day)
-            logger.info(f"{day} is regridded")
-            interpolated = interpolate_nans(var, day)
-            logger.info(f"{day} is interpolated")
-            masked = mask(interpolated)
-            logger.info(f"{day} is masked")
-            masked.to_netcdf(args.out_path + var + "_" + day + ".nc")
-            logger.info(f"{day} is done")
+        merge_into_weekly_file(path, var, args)
+        logger.info(f"{var} is merged into weekly file")
+        regrid_to_cubed_sphere(path, var)
+        logger.info(f"{var} is regridded")
+        interpolated = interpolate_nans(var)
+        interpolated.to_netcdf("interpolated.nc")
+        logger.info(f"{var} is interpolated")
+        masked = mask(interpolated)
+        masked.to_netcdf("temp.nc")
+        logger.info(f"{var} is masked")
+        if var in ["temp2m", "u_wind", "v_wind"]:
+            shifted = shift_variable(masked)
+        else:
+            shifted = masked
+        shifted.to_netcdf(
```

Comment:
There's a number of temporary netcdfs that are written in this script before the final merged output is saved. For the intermediate netcdfs that are not needed, I would create them in a[ temporary directory](https://stackoverflow.com/questions/3223604/how-do-i-create-a-temporary-directory-in-python) so they are cleaned up at the end.

Reply:
I tried this, but it caused many issues with making directories or writing files, so I just created a /temp folder and removed it after